airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/ssh/SshHelpers.java

```diff
@@ -12,30 +13,41 @@
 import java.util.Optional;
 
 public class SshHelpers {
+  private static final String SSH_KEY_PATTERN = "-----BEGIN[ a-zA-Z0-9]+-----(.|\\n)*-----END[ a-zA-Z0-9]+-----";
+  private static final String SSH_KEY_PATTERN_DESCRIPTOR = "PEM format";
 
   public static ConnectorSpecification getSpecAndInjectSsh() throws IOException {
-    return getSpecAndInjectSsh(Optional.empty());
+    return getSpecAndInjectSsh(Optional.empty(), false);
   }
 
-  public static ConnectorSpecification getSpecAndInjectSsh(final Optional<String> group) throws IOException {
+  public static ConnectorSpecification getSpecAndInjectSsh(final Optional<String> group, final boolean setSshKeyPatterns) throws IOException {
     final ConnectorSpecification originalSpec = Jsons.deserialize(MoreResources.readResource("spec.json"), ConnectorSpecification.class);
-    return injectSshIntoSpec(originalSpec, group);
+    return injectSshIntoSpec(originalSpec, group, setSshKeyPatterns);
   }
 
   public static ConnectorSpecification injectSshIntoSpec(final ConnectorSpecification connectorSpecification) throws IOException {
-    return injectSshIntoSpec(connectorSpecification, Optional.empty());
+    return injectSshIntoSpec(connectorSpecification, Optional.empty(), false);
   }
 
-  public static ConnectorSpecification injectSshIntoSpec(final ConnectorSpecification connectorSpecification, final Optional<String> group)
+  public static ConnectorSpecification injectSshIntoSpec(final ConnectorSpecification connectorSpecification, final Optional<String> group, final boolean setSshKeyPatterns)
       throws IOException {
     final ConnectorSpecification originalSpec = Jsons.clone(connectorSpecification);
     final ObjectNode propNode = (ObjectNode) originalSpec.getConnectionSpecification().get("properties");
     final ObjectNode tunnelMethod = (ObjectNode) Jsons.deserialize(MoreResources.readResource("ssh-tunnel-spec.json"));
     if (group.isPresent()) {
       tunnelMethod.put("group", group.get());
     }
+    if (setSshKeyPatterns) {
+      tunnelMethod.get("oneOf").elements().forEachRemaining(node -> {
+        final JsonNode properties = node.get("properties");
+        if (properties.get("tunnel_method").get("const").asText().equals("SSH_KEY_AUTH")) {
+          final ObjectNode sshKey = (ObjectNode) node.get("properties").get("ssh_key");
+          sshKey.put("pattern", SSH_KEY_PATTERN);
+          sshKey.put("pattern_descriptor", SSH_KEY_PATTERN_DESCRIPTOR);
+        }
+      });
+    }
```

Comment:
@rodireich I wanted to limit the effect of this change to just the postgres connector, so I couldn't simply add these fields to the `ssh-tunnel-spec.json` file since that is consumed by a lot of different connectors.

I therefore opted to add this logic to set these ssh key patterns if a new `setSshKeyPatterns` argument is set to true, which currently only the Postgres source connector is setting.

Is this approach okay to you? Alternatively we could also create a new `ssh-tunnel-spec-postgres.json` file with these things set directly, and remove the conditional logic in this class. But that would mean we need to duplicate most of the `ssh-tunnel-spec.json` file, and that could cause the two to drift.

Reply:
This sounds good - better not create a duplicate.
Just one question: why is this better approach than editing the spec of source postgres? Isn't that isolated to just that connector as well?

Reply:
@rodireich the `tunnel_method` property doesn't exist in the source-postgres spec, because the ssh tunnel spec fields get dynamically inserted into the postgres spec by this function from the `ssh-tunnel-spec.json` file. So there is nowhere in the source-postgres spec to set these patterns

Reply:
got it. thanks for clarifying